packages/block-editor/src/components/rich-text/index.js

```diff
@@ -154,15 +158,22 @@ export function RichTextWrapper(
 			selectionStart: isSelected ? selectionStart.offset : undefined,
 			selectionEnd: isSelected ? selectionEnd.offset : undefined,
 			isSelected,
+			isWriteMode:
+				window?.__experimentalEditorWriteMode &&
+				isNavigationMode() &&
```

Comment:
I think the `isNavigationMode` already checks the `window?.__experimentalEditorWriteMode`.